examples/js/hello-world/app.js

```diff
@@ -0,0 +1,139 @@
+                bindings: [
+                    {
+                        location: 'send-button',
+                        icon: 'http://localhost:8080/static/icon.png',
+                        label: 'send hello message',
+                        call: {
+                            path: '/send-modal',
+                        },
+                    },
+                ],
+            },
+            {
+                location: '/command',
+                bindings: [
+                    {
+                        icon: 'http://localhost:8080/static/icon.png',
+                        label: 'helloworld',
+                        description: 'Hello World app',
+                        hint: '[send]',
+                        bindings: [
+                            {
+                                location: 'send',
+                                label: 'send',
+                                call: {
+                                    path: '/send',
+                                },
+                            },
+                        ],
+                    },
+                ],
+            },
+        ],
+    });
+});
+
+app.post(['/send/form', '/send-modal/submit'], (req, res) => {
+    res.json({
+        type: 'form',
+        form: {
+            title: 'Hello, world!',
+            icon: 'http://localhost:8080/static/icon.png',
+            fields: [
+                {
+                    type: 'text',
+                    name: 'message',
+                    label: 'message',
+                },
+            ],
+            call: {
+                path: '/send',
+            },
+        },
+    });
+});
+
+app.get('/static/icon.png', (req, res) => {
+    res.sendFile(__dirname + '/icon.png');
+});
+
+app.use(express.json());
+
+app.post('/send/submit', (req, res) => {
+    const call = req.body;
+
+    let message = 'Hello, world!';
+    const user_message = call.values.message;
+    if (user_message) {
+        message += ' ...and ' + user_message + '!';
+    }
+
+    const users = [
+        call.context.bot_user_id,
+        call.context.acting_user_id,
+    ];
+
+    // Use the app bot to do API calls
+    const options = {
+        method: 'POST',
+        headers: {
+            Authorization: 'BEARER ' + call.context.bot_access_token,
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(users),
+    };
+
+    // Get the DM channel between the user and the bot
+    const mattermost_site_url = call.context.mattermost_site_url;
+
+    fetch(mattermost_site_url + '/api/v4/channels/direct', options).
+        then((mm_res) => mm_res.json()).
+        then((channel) => {
+            const post = {
+                channel_id: channel.id,
+                message,
+            };
+
+            // Create a post
+            options.body = JSON.stringify(post);
+
+            fetch(mattermost_site_url + '/api/v4/posts', options);
```

Comment:
No error handling here?

Reply:
Also, I think it would be helpful to have the example return `{markdown: 'Some ephemeral response'}` as it is very easy for the developer to implement, and showcases something specific to the framework. This can be in addition to the public post made here. I'm aware that the Go example does not make use of the ephemeral post.

If we remove the public post created here and use the `markdown` option to create an ephemeral post instead, the whole example becomes much simpler too.

Reply:
Hello apps generally skip on error handling to improve the readability.

Perhaps comments are warranted everywhere, and 2/5 maybe a separate `hello-errors?` @justinegeffen

Reply:
@mickmister I like the public post because it shows the use of the bot access token. 2/5 to keep it. In terms of error handling: I agree with @levb that it doesn't make the example more readable.

Reply:
Can we also include the `markdown` response (in one of the handlers), or is that not applicable here? I just think it's definitely something we want to emphasize because it's available and very easy to implement on the app's side.

Reply:
Good point. I've added a `markdown` response in be76159.